BAZEL-JVM.md

```diff
@@ -106,212 +106,68 @@ Transitive dependencies are introduced implicitly as dependencies of direct
 dependencies. It is preferable to only have to define direct dependencies and
 let a dependency resolver determine all transitive dependencies.
 
-The tool [`bazel-deps`][bazel-deps] can read a list of direct Java and Scala
-JAR dependencies and then use [Coursier][coursier] to perform dependency
-resolution and generate the whole transitive closure of dependencies. The
-`dev-env` provides a modified version of `bazel-deps` that supports Artifactory
-authentication as required at Digital Asset.
-
-Please do not execute `bazel-deps` directly. Instead, use the
-`update-bazel-deps` tool provided in the dev-env. Make sure to check-in the
-resulting files into revision control.  
-
-Direct dependencies are manually defined in the file `dependencies.yaml` at the
-repository root. The generated Bazel definitions are written to the `3rdparty`
-directory in the repository root. The file `3rdparty/workspace.bzl` lists the
-individual JAR dependencies and the directory tree under `3rdparty/jvm` defines
-Bazel targets for them. Only Bazel targets for direct dependencies are publicly
-visible. The Bazel targets include `exports` attributes to capture indirect
-dependencies.
+The [`rules_jvm_external`][rules_jvm_external] rules set can read a list of
+direct JAR dependencies and then use [Coursier][coursier] to perform dependency
+resolution and generate the whole transitive closure of dependencies.
 
-The file `3rdparty/workspace.bzl` defines Bazel macros that must be called in
+Direct dependencies are manually defined in the file `bazel-java-deps.bzl` at
+the repository root. This file also defines Bazel macros that must be called in
 the `WORKSPACE` file in order to import the external JAR dependencies.
 
 ```
-load("//3rdparty:workspace.bzl", "maven_servers", "maven_dependencies")
-maven_servers()
-maven_dependencies()
+load("//:bazel-java-deps.bzl", "install_java_deps")
+install_java_deps()
+load("@maven//:defs.bzl", "pinned_maven_install")
+pinned_maven_install()
 ```
 
-First, we call `maven_servers` which will define the list of repositories from
-which to fetch JARs. These are listed in the definition of `list_servers` in
-`3rdparty/workspace.bzl`. For example:
+The macro `install_java_deps` defines the direct dependencies and the tools to
+load and update pinned versions. The macro `pinned_maven_install` loads the
+pinned artifacts into Bazel. Within `bazel-java-deps.bzl` we call
+`maven_install` where we define all direct Maven dependencies.
 
 ```
-{"name": "central", "url": "https://digitalasset.jfrog.io/digitalasset/libs-release"},
+maven_install(
+    artifacts = [
+        "com.google.guava:guava:24.0-jre",
+        "org.scalaz:scalaz-core_2.12:7.2.24",
+        ...
+    ],
```

Comment:
```suggestion
    ],)
```

Reply:
Good catch, `maven_install` takes further attributes, so I've changed it to
```
maven_install(
    artifacts = [
        "com.google.guava:guava:24.0-jre",
        "org.scalaz:scalaz-core_2.12:7.2.24",
        ...
    ],
    ...
)